src/rp2_common/hardware_sync_spin_lock/include/hardware/sync/spin_lock.h

```diff
@@ -19,53 +19,93 @@
 
 // PICO_CONFIG: PICO_SPINLOCK_ID_IRQ, Spinlock ID for IRQ protection, min=0, max=31, default=9, group=hardware_sync
 #ifndef PICO_SPINLOCK_ID_IRQ
+#if PICO_RP2350 && !PICO_USE_SW_SPIN_LOCKS
+#define PICO_SPINLOCK_ID_IRQ 5
+#else
 #define PICO_SPINLOCK_ID_IRQ 9
 #endif
+#endif
 
 // PICO_CONFIG: PICO_SPINLOCK_ID_TIMER, Spinlock ID for Timer protection, min=0, max=31, default=10, group=hardware_sync
 #ifndef PICO_SPINLOCK_ID_TIMER
+#if PICO_RP2350 && !PICO_USE_SW_SPIN_LOCKS
+#define PICO_SPINLOCK_ID_TIMER 6
+#else
 #define PICO_SPINLOCK_ID_TIMER 10
 #endif
+#endif
```

Comment:
RP2350-E2 says that 10 is a "safe" spinlock register, so is there any reason to re-define spinlock 10 as `PICO_SPINLOCK_ID_RAND` rather than leaving it defined as `PICO_SPINLOCK_ID_TIMER` ?

Reply:
I thought it better to keep them in the same order